### PR TITLE
Formal professional services

### PR DESCRIPTION
**Pull request by rainbreak**
previous latest run from https://github.com/dapphub/k-dss/pull/39: https://dapp.ci/k-dss/4296d527e38631f8b756/

796/812 accepted proofs.

Remaining problem areas:

- vat lemmas for fold/grab/frob/fork fail cases
- skip/bite pass cases
- end.cage k performance (oom currently)
- drip fail cases are entering rpow loop and timing out
- flapper.tend fail case timing out

### CODE REVIEW

File: src/dirty_lemmas.k.md [hunk 1]
```diff
@@ -141,9 +141,9 @@ rule #unsigned(X) -Int (pow256 +Int X) => 0
 
 // Cat_bite_full_fail_rough lemma
 rule #sgnInterp(sgn(chop(A *Int #unsigned(0 -Int B))) *Int -1, abs(chop(A *Int #unsigned(0 -Int B))) /Int B) ==K A => #rangeSInt(256, A *Int (0 -Int B))
-  requires #rangeUInt(256, A)
-  andBool 0 <Int B
-  andBool #rangeSInt(256, 0 -Int B)
+//   requires #rangeUInt(256, A)
```
Comment (MrChico) [inline src/dirty_lemmas.k.md:144]: danger zone

File: src/dirty_lemmas.k.md [hunk 2]
```diff
@@ -141,9 +141,9 @@ rule #unsigned(X) -Int (pow256 +Int X) => 0
 
 // Cat_bite_full_fail_rough lemma
 rule #sgnInterp(sgn(chop(A *Int #unsigned(0 -Int B))) *Int -1, abs(chop(A *Int #unsigned(0 -Int B))) /Int B) ==K A => #rangeSInt(256, A *Int (0 -Int B))
- requires #rangeUInt(256, A)
- andBool 0 <Int B
- andBool B <=Int pow255
+  requires #rangeUInt(256, A)
+  andBool 0 <Int B
```
Comment (MrChico) [inline src/dirty_lemmas.k.md]: this is the problem. Changing the next line doesn't do anything,..

**Comment by MrChico** [comment]
I have corrected the `mul` lemma me and @livnev found a problem with yesterday:
```
444c444
< rule A *Int #unsigned(B) => #unsigned(A *Int B)
---
> rule A *Word #unsigned(B) => #unsigned(A *Int B)
```
but since it is in `lemmas.k.md` this will trigger rerun (and we haven't tested whether this causes problems)

**Comment by mhhf** [comment]
@rainbreak something we changed yesterday let us make progress with cat bite, however - the constraints seems to be codependent :(

**Comment by livnev** [comment]
@rainbreak @MrChico  can this be merged?

**Comment by MrChico** [comment]
Fine by me

**Comment by gbalabasquer** [comment]
Guys can this be merged, please?

**Review by MrChico — APPROVED** [review]

**Comment by gbalabasquer** [comment]
Thanks!

**Pull request merged**